packages/turf-clusters-kmeans/index.js

```diff
@@ -27,37 +23,36 @@ var featureCollection = helpers.featureCollection;
  * var clustered = turf.clustersKmeans(points, numberOfClusters);
  *
  * //addToMap
- * var addToMap = featureCollection(clustered.points);
+ * var addToMap = clustered;
  */
 module.exports = function (points, numberOfClusters) {
     // Input validation
     collectionOf(points, 'Point', 'Input must contain Points');
+
     // Default Params
     var count = points.features.length;
-    if (numberOfClusters > count) throw new Error('numberOfClusters can\'t be grated than the number of points');
     numberOfClusters = numberOfClusters || Math.round(Math.sqrt(count / 2));
 
+    // numberOfClusters can't be greater than the number of points
+    // fallbacks to count
+    if (numberOfClusters > count) numberOfClusters = count;
+
+    // Clone points to prevent any mutations
+    points = clone(points, true);
+
     // collect points coordinates
-    var data = [];
-    coordEach(points, function (coord) {
-        data.push(coord);
-    });
+    var data = coordAll(points);
 
     // create seed to avoid skmeans to drift
     var initialCentroids = data.slice(0, numberOfClusters);
 
     // create clusters
-    var clastersResult = skmeans(data, numberOfClusters, initialCentroids);
-    var centroids = [];
-    clastersResult.centroids.forEach(function (coord, idx) {
-        centroids.push(point(coord, {cluster: idx}));
-    });
-    featureEach(points, function (pt, i) {
-        pt.properties.cluster = clastersResult.idxs[i];
+    var clustered = skmeans(data, numberOfClusters, initialCentroids);
+
+    // add associated cluster number
+    featureEach(points, function (point, index) {
+        point.properties.cluster = clustered.idxs[index];
     });
 
-    return {
-        points: points,
-        centroids: featureCollection(centroids)
-    };
+    return points;
```

Comment:
I see the reasons why we want to output GeoJSON objects, however I believe the centroids in this particular clustering method are relevant and would be beneficial to the user having them already available in the output.
Couldn't we add them as a `centroids` property of the output? It would still be a valid GeoJSON format